### DISCUSSION TITLE
Custom class for nodes

### DISCUSSION THREAD

**Discussion by endmax** [Blazor-Diagrams/Blazor.Diagrams#294] [General]
I need to style nodes in different ways depending on my business rules.
The following image shows a simple diagram I need to generate and manage.
![image](https://user-images.githubusercontent.com/2227603/209991016-97509b44-2b89-4d56-8a12-d7344b99db91.png)

So I've used blazor.diagram but I've found no way to simply obtain such a result other than modifying the code in this way:

NodeWidget.razor
```
<div class="default-node @(Node.Group != null ? "grouped" : "") @(Node.Selected ? "selected" : "") @(string.IsNullOrEmpty(Node.CustomClass) ? "" : Node.CustomClass)">
    @(Node.Title ?? "Title")
    @foreach (var port in Node.Ports)
    {
        <PortRenderer Port="port" Class="default"></PortRenderer>
    }
</div>
```
NodeModel.cs
`public string CustomClass { get; set; }`

Any suggestion ?

**Comment by zHaytam** [comment]
Hello,
The optimal solution would be to have 1 separate model (class) for each node type, and you assign 1 widget to each one and you style it however you want.
If the only difference is the background color for example, a property in one class is enough in my opinion.
